File: src/main/java/com/optica/repository/login/UserRepository.java
package com.optica.repository.login;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import com.optica.entity.User;


@Repository
public interface UserRepository extends JpaRepository<User, Long> {
	
	User findFirstByUsername(String username);

}

File: src/main/kotlin/com/optica/repository/login/UserRepository.kt
package com.optica.repository.login

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

import com.optica.entity.User


@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun findFirstByUsername(username: String?): User

}
